### DISCUSSION TITLE
Eslint: require-jsdoc deprecated in ESLint v5

### DISCUSSION THREAD

**Discussion by juliankasimir** [eviden-parallel/community#16] [General]
Actual we use jsdoc requirments in Eslint, but it would be [deprecated ](https://eslint.org/docs/latest/rules/require-jsdoc)in v5.

.eslintrc:
```
"valid-jsdoc": "error",
```

```
"require-jsdoc": ["error",{ "require": { "FunctionDeclaration": true, "MethodDefinition": true, "ClassDeclaration": true } }],
```

**Comment by juliankasimir** [comment]
Solution could be to use [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc):

.eslintrc.json:
```
  "extends": [
    "eslint:recommended",
    "plugin:jsdoc/recommended-error",
    "plugin:jsdoc/recommended-typescript-error",
    "prettier"
  ],
```

```
    {
      "extends": [
        "plugin:jsdoc/recommended-error",
        "plugin:jsdoc/recommended-typescript-error"
      ],
      "files": ["*.{js,ts}"],
      "rules": {
        "jsdoc/check-indentation": 1,
        "jsdoc/check-line-alignment": 1,
        "jsdoc/check-param-names": 1,
        "jsdoc/check-tag-names": 1,
        "jsdoc/informative-docs": 1,
        "jsdoc/multiline-blocks": 1,
        "jsdoc/no-multi-asterisks": 1,
        "jsdoc/require-asterisk-prefix": 1,
        "jsdoc/require-description": 1,
        "jsdoc/require-jsdoc": 1,
        "jsdoc/valid-types": 1
      }
    }
```

```
  "plugins": [
    "jsdoc"
  ],
```

package.json:
`"eslint-plugin-jsdoc": "^46.4.4",`

**Comment by Th3S4mur41** [comment]
Since the plan is to move to rome which currently neither includes jsdoc rules nor supports plugin.
Adding the plugin now just to remove it again later makes little sense.

Moreover, appart for a few exceptions (e.g. some few scripts and helper classes) most repositories define very few functions and the ones we write should be documented in storybook anyway.
So, in this project the benefit of enforcing jsdoc is not that big.

Which doesn't mean in any way, that we shouldn't write jsdoc when relevant
